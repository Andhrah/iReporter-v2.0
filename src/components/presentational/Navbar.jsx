/* eslint-disable react/jsx-no-bind */
import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import logo from '../../../public/images/logo.png';
import toggle from '../../utils/navbarToggle';

const Navbar = ({
  firstname, lastname, username, email
}) => (
  <div className="wrapper">
    <header>
      <div className="branding">
        <Link to="/">
          <img src={logo} alt="iReporter logo" />
        </Link>
      </div>
      <nav className="navbar" id="topnav">
        <ul>
          <li><Link to="/redflags">Red-flags</Link></li>
          <li><Link to="/interventions">Interventions</Link></li>
          <li><Link to="/create-redflag">Report Corruption</Link></li>
          <li><Link to="/create-intervention">Call for Intervention</Link></li>
          {firstname && lastname && username && email ? (
            <Fragment>
              {/* <span>
                Welcome
                {' '}
              </span> */}
              <span>
                {firstname}
                {' '}
              </span>
              <span>
                <button
                  type="button"
                  className="logout"
                  onClick={() => {
                    localStorage.clear();
                    window.location.href = '/';
                  }}>
                    LOG OUT
                </button>

              </span>
            </Fragment>
          ) : (
            <Fragment>
              <li>
                <Link to="/signup">
                    Sign Up
                  <i className="fa fa-user-plus" />
                </Link>
              </li>
              <li>
                <Link to="/login">
                    Sign In
                  {' '}
                  <i className="fa fa-user" />
                </Link>
              </li>
            </Fragment>
          )
          }
          <li>
            <button
              type="button"
              // eslint-disable-next-line no-script-url
              to="javascript:void(0);"
              className="icon"
              onClick={toggle}>
              <i className="fa fa-bars" />
            </button>
          </li>
        </ul>
      </nav>
    </header>
  </div>

);


Navbar.propTypes = {
  firstname: PropTypes.string,
  lastname: PropTypes.string,
  username: PropTypes.string,
  email: PropTypes.string,
};

Navbar.defaultProps = {
  firstname: '',
  lastname: '',
  username: '',
  email: '',
};

export default Navbar;
