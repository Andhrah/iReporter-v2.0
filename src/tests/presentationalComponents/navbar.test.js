import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../../components/presentational/Navbar';

const props = {
  firstname: 'Andra',
  lastname: 'Collins',
  username: 'Alexandra',
  email: 'alexandra@test.com',
};

describe('Navbar test suite', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Navbar {...props} />
      </MemoryRouter>
    );
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('logs a user out', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Navbar {...props} />
      </MemoryRouter>
    );
    wrapper.find('.logout').simulate('click');
    expect(wrapper.instance().props.firstname).toEqual(undefined);
  });
});
