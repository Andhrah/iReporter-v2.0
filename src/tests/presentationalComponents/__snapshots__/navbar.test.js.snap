// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar test suite renders correctly 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "8julib",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "8julib",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Navbar
      email="alexandra@test.com"
      firstname="Andra"
      lastname="Collins"
      username="Alexandra"
    >
      <div
        className="wrapper"
      >
        <header>
          <div
            className="branding"
          >
            <Link
              to="/"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                <img
                  alt="iReporter logo"
                  src={Object {}}
                />
              </a>
            </Link>
          </div>
          <nav
            className="navbar"
            id="topnav"
          >
            <ul>
              <li>
                <Link
                  to="/redflags"
                >
                  <a
                    href="/redflags"
                    onClick={[Function]}
                  >
                    Red-flags
                  </a>
                </Link>
              </li>
              <li>
                <Link
                  to="/interventions"
                >
                  <a
                    href="/interventions"
                    onClick={[Function]}
                  >
                    Interventions
                  </a>
                </Link>
              </li>
              <li>
                <Link
                  to="/create-redflag"
                >
                  <a
                    href="/create-redflag"
                    onClick={[Function]}
                  >
                    Report Corruption
                  </a>
                </Link>
              </li>
              <li>
                <Link
                  to="/create-intervention"
                >
                  <a
                    href="/create-intervention"
                    onClick={[Function]}
                  >
                    Call for Intervention
                  </a>
                </Link>
              </li>
              <span>
                Andra
                 
              </span>
              <span>
                <button
                  className="logout"
                  onClick={[Function]}
                  type="button"
                >
                  LOG OUT
                </button>
              </span>
              <li>
                <button
                  className="icon"
                  onClick={[Function]}
                  to="javascript:void(0);"
                  type="button"
                >
                  <i
                    className="fa fa-bars"
                  />
                </button>
              </li>
            </ul>
          </nav>
        </header>
      </div>
    </Navbar>
  </Router>
</MemoryRouter>
`;
